src/parser.rs

```diff
@@ -1208,8 +1208,102 @@ impl<'a> Parser<'a> {
                 out.push(self.parse_statement(line_raw, constructor, &mut stack, &mut iterator));
             }
         }
+        self.check_statements(out.clone())
```

Comment:
this is actually the same way, we still need to iterate over all statements again

instead of push (line 1208) check if the statement is catch/else/else if

if yes, check if the last in out is comment, and while yes, pop them to another vec, then push them to the parsed statement statements (we know it is a block)

and then push to out

Reply:
yes, I use this algorithm in one fn and forgot for another one, my bad